src/components/hooks/HookDetails.tsx

```diff
@@ -0,0 +1,185 @@
+  let prettyHookInput = JSON.stringify(hookInput, null, 2);
+
+  let hookOutput = JSON.parse(props.hook.info.result);
+  let prettyHookOutput = JSON.stringify(hookOutput, null, 2);
+
+  // Extract hook-specific data
+  let targetName;
+  let targetState;
+
+  switch (hook.name) {
+    case 'on_task':
+      targetName = 'Task';
+      targetState = hookInput[0].payload.data.task.status;
+      break;
+    case 'on_build':
+      targetName = 'Build';
+      targetState = hookInput[0].payload.data.build.status;
+      break;
+    default:
+      targetName = 'Unknown';
+      targetState = 'UNKNOWN';
+      break;
+  }
+
+  // In case this is a task hook — display a chip that links to it
+  const taskChip =
+    hook.task != null ? <TaskNameChip className={classes.chip} task={hook.task} withNavigation={true} /> : null;
+
+  // Potential error from the hook execution
+  let headerStyle = {
+    backgroundColor: useNotificationColor('ERROR'),
+  };
+  const potentialError =
+    hook.info.error === '' ? null : (
+      <div className={classNames('container', classes.gap)}>
+        <div
+          key={props.hook.info.error}
+          style={headerStyle}
+          className={classNames('row', 'justify-content-between', 'align-items-center')}
+        >
+          <Typography variant="subtitle1" className={classes.potentialError}>
+            {props.hook.info.error}
+          </Typography>
+        </div>
+      </div>
+    );
+
+  return (
+    <div>
+      <CirrusFavicon status={hook.info.error === ''} />
+      <Head>
+        <title>{targetName} hook - Cirrus CI</title>
+      </Head>
+      <Card>
+        <CardContent>
+          <div className={classes.wrapper}>
+            <RepositoryNameChip className={classes.chip} repository={hook.repository} />
+            <BuildBranchNameChip className={classes.chip} build={hook.build} />
+            <BuildChangeChip className={classes.chip} build={hook.build} />
+            {taskChip}
+          </div>
+          <div className={classes.wrapper}>
+            <HookCreatedChip className={classes.chip} hook={hook} />
+            <HookStatusChip className={classes.chip} hook={hook} />
+          </div>
+          <div className={classes.gap} />
+          <Typography variant="h6" gutterBottom>
+            {targetName} hook
+          </Typography>
+          <p>
+            Triggered because <code>{hook.name}()</code> was defined and the has task entered the{' '}
+            <code>{targetState}</code> state.
+          </p>
+        </CardContent>
+      </Card>
+      {potentialError}
+      <div className={classes.gap} />
+      <Card>
+        <CardContent>
+          <Typography variant="h6">Execution logs</Typography>
+          <Logs logsName="output" logs={props.hook.info.outputLogs.join('\n')} />
+        </CardContent>
+      </Card>
+      <div className={classes.gap} />
+      <Card>
+        <CardContent>
+          <Typography variant="h6">Input</Typography>
```

Comment:
Maybe call it `Arguments` since theses are the arguments to the hook function.

Reply:
See https://github.com/cirruslabs/cirrus-ci-web/pull/355/commits/0d59f701009e007b615f382534e01f95db5cea72.